resteasy-core/src/main/java/org/jboss/resteasy/core/registry/RootNode.java

```diff
@@ -56,10 +69,12 @@ public ResourceInvoker match(HttpRequest request, int start)
          request.setAttribute(RESTEASY_CHOSEN_ACCEPT, match.chosen);
       } else {
          match = root.match(request, start);
-         if (match.match != null && match.match.expression.getNumGroups() == 0 && match.invoker instanceof ResourceMethodInvoker) {
+         if (cache.size() < CACHE_SIZE && match.match != null && match.match.expression.getNumGroups() == 0 && match.invoker instanceof ResourceMethodInvoker) {

```

Comment:
So once the cache is full, nothing gets added / old entries removed?

Reply:
It's ideal that we remove the LRU entry when the cache is full.  Later we can look at [ConcurrentLinkedHashMap](https://github.com/ben-manes/concurrentlinkedhashmap)  which Ben suggested.

Reply:
Just remember, that 99% of apps will only have a handful of media types (< 5), usually only 1 (json).  Might be a little bigger if character sets are involved, but there's only a handful of those in existence as well.  The issue here was a media type that adds a random parameter.  IMO, if you hit the cache size you're dealing with an app that has another random parameter and the cache will be useless.  Might as well clear it and turn it off in that scenario.

Reply:
Isn't this per route? The cache size would be hit if an application has more than `CACHE_SIZE` routes, in the expected case.

The `MediaTypeHeaderDelegate` is a cache of media types and it clears itself when full. It doesn't condition for dynamic parameters, so a multipart route would invalidate it regularly.